core/src/main/scala/kafka/controller/ZookeeperClient.scala

```diff
@@ -145,26 +153,59 @@ class ZookeeperClient(connectString: String, sessionTimeoutMs: Int, connectionTi
     info("Connected.")
   }
 
+  /**
+   * Register the handler to ZookeeperClient. This is just a local operation. This does not actually register a watcher.
+   *
+   * The watcher is only registered once the user calls handle(AsyncRequest) or handle(Seq[AsyncRequest])
+   * with either a GetDataRequest or ExistsRequest.
+   *
+   * NOTE: zookeeper only allows registration to a nonexistent znode with ExistsRequest.
+   *
+   * @param zNodeChangeHandler the handler to register
+   */
   def registerZNodeChangeHandler(zNodeChangeHandler: ZNodeChangeHandler): Unit = {
     registerZNodeChangeHandlers(Seq(zNodeChangeHandler))
   }
 
+  /**
+   * Register the handlers to ZookeeperClient. See registerZNodeChangeHandler(ZNodeChangeHandler) for details.
+   * @param handlers the handlers to register
+   */
   def registerZNodeChangeHandlers(handlers: Seq[ZNodeChangeHandler]): Unit = {
     handlers.foreach(handler => zNodeChangeHandlers.put(handler.path, handler))
   }
 
+  /**
+   * Unregister the handler from ZookeeperClient. This is just a local operation.
+   * @param path the path of the handler to unregister
+   */
   def unregisterZNodeChangeHandler(path: String): Unit = {
     zNodeChangeHandlers.remove(path)
   }
 
+  /**
+   * Register the handler to ZookeeperClient. This is just a local operation. This does not actually register a watcher.
+   *
+   * The watcher is only registered once the user calls handle(AsyncRequest) or handle(Seq[AsyncRequest]) with a GetChildrenRequest.
+   *
+   * @param zNodeChildChangeHandler the handler to register
+   */
   def registerZNodeChildChangeHandler(zNodeChildChangeHandler: ZNodeChildChangeHandler): Unit = {
     registerZNodeChildChangeHandlers(Seq(zNodeChildChangeHandler))
   }
 
+  /**
+   * Register the handlers to ZookeeperClient. See registerZNodeChildChangeHandler(ZNodeChildChangeHandler) for details.
```

Comment:
Now that we've decoupled handler registration from watcher registration, there's really no benefit to having `registerZNodeChangeHandlers` and `registerZNodeChildChangeHandlers` since these are now purely local operations and equivalent to registering one-at-a-time.

I'm going to remove these methods.